sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc

```diff
@@ -80,9 +80,61 @@ properties P1{bar};
 properties P2{bar, foo{argv[0]}};
 properties P3{bar, baz};
 properties P4{baz, bar};
+
 static_assert(!std::is_same_v<decltype(P1), decltype(P2)>); // P1 and P2 contain different properties
 static_assert(!std::is_same_v<decltype(P1), decltype(P3)>); // P1 and P3 contain different properties
 static_assert(std::is_same_v<decltype(P3), decltype(P4)>); // The order properties are specified doesn't change the type
+
+```
+
+The alias `sycl::ext::oneapi::experimental::properties_t` is defined for
+convenience and it is defined as follows:
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+template <auto... Props> using properties_t = decltype(properties{Props...});
+
+} // namespace experimental::oneapi::ext::sycl
+```
+
+Use this alias in situations where you need to get the type of a `properties`
+object that consists of 1 or more compile time constant properties. For example
+consider the following toy examples:
+
+.Examples using decltype
+```c++
+
+// buffer_location and awidth are some compile time constant properties
+annotated_arg<int *, decltype(properties{buffer_location<1>, awidth<32>})> a;
+
+// protocol_avalon_mm and uses_ready<false> are some compile time constant
+// properties
+using Pipe = pipe<..., decltype(properties{protocol_avalon_mm,
+                                           uses_ready<false>})>;
+
+// device_image_scope and host_access_read_write are some compile time constant
+// properties
+device_global<int, decltype(properties{device_image_scope,
+                                       host_access_read_write})> global_obj;
+```
+
+Such use cases arise often with SYCL extensions that are built to utilize
+`properties`. The above can be simplified to:
+
+.Examples using properties_t
+```c++
+using namespace sycl::ext::oneapi::experimental;
+
+// buffer_location and awidth are compile time constant properties
+annotated_arg<int *, properties_t<buffer_location<1>, awidth<32>> a;
+
+using Pipe = pipe<..., properties_t<protocol_avalon, uses_ready<false>>>;
+
+// device_image_scope and host_access_read_write are some compile time constant
+// properties
+device_global<int, properties_t<device_image_scope, host_access_read_write>>
+  global_obj;
```

Comment:
I don't think we want to show examples of other extensions like `pipe` or `device_global` here.  Instead, update those extension specs to use `properties_t` instead of `decltype`.  Those changes can be part of this same PR.

Reply:
I think we will handle that in the FPGA docs directly. Updated to remove these examples.